app/models/runs_extractors.rb

```diff
@@ -60,15 +64,31 @@ def extract(classification, and_reduce: false)
 
     if and_reduce
       extracts = extracts.select { |extract| extract != Extractor::NoData }
+      return unless extracts.present?
 
-      if extracts.present?
-        ids = extracts.map(&:id)
-        ReduceWorker.perform_async(classification.workflow_id, "Workflow", classification.subject_id, classification.user_id, ids)
+      ids = extracts.map(&:id)
 
-        project = Project.find_by_id(workflow.project_id)
-        if project && project.has_reducers?
-          ReduceWorker.perform_async(classification.project_id, "Project", classification.subject_id, classification.user_id, ids)
+      if workflow && workflow.reducers.any?
+        worker = if workflow.has_external_reducers?
+          ReduceWorkerExternal
+        else
+          ReduceWorker
         end
+
+        worker.set(queue: workflow.custom_queue_name.to_sym) if workflow.custom_queue_name.present?
```

Comment:
should the queue names be strings or symbols?

https://github.com/mperham/sidekiq/wiki/Advanced-Options#workers

I see the docs use symbols but they are user generated, have we tested this end to end with a real working system to ensure that jobs flow correctly?

Maybe worth ensuring consistency between DSL methods on the worker `sidekiq_options` and this dynamic setting https://github.com/mperham/sidekiq/blob/dea31322c4489bd2c1b7cc9ce822e945fa0e9e8b/lib/sidekiq/worker.rb#L149

and looks like it is taken care for us but worth thinking about, https://github.com/mperham/sidekiq/blob/d16572bcdd0aa52985f9dc9e79f5179a6c828154/lib/sidekiq/client.rb#L60

https://github.com/mperham/sidekiq/blob/d16572bcdd0aa52985f9dc9e79f5179a6c828154/lib/sidekiq/client.rb#L227

Reply:
since they're normalized in the library code, i prefer using symbols for better readability / code completion in our consumer code, but i'm not religious about it